### PR TITLE
8293680: PPC64BE build failure after JDK-8293344

### PR DESCRIPTION
**Pull request by shipilev**
The bug manifests on PPC64, since `section_by_name` is protected by `#if defined(PPC64) && !defined(ABI_ELFv2)`:

```
* For target hotspot_variant-server_libjvm_objs_elfFile.o:
/home/shade/trunks/jdk/src/hotspot/share/utilities/elfFile.cpp: In member function 'int ElfFile::section_by_name(const char*, Elf_Shdr&)':
/home/shade/trunks/jdk/src/hotspot/share/utilities/elfFile.cpp:261:16: error: aggregate 'ResourceMark rm' has incomplete type and cannot be defined
  261 | ResourceMark rm;
      | ^~
``` 

This restores the include that previous commit removed: https://github.com/openjdk/jdk/commit/d4e3e69505db1b114afec2f6a61acf1261a8e69c

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8293680](https://bugs.openjdk.org/browse/JDK-8293680): PPC64BE build failure after JDK-8293344

### Reviewers
 * [Martin Doerr](https://openjdk.org/census#mdoerr) (@TheRealMDoerr - **Reviewer**)
 * [Thomas Stuefe](https://openjdk.org/census#stuefe) (@tstuefe - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk pull/10247/head:pull/10247` \
`$ git checkout pull/10247`

Update a local copy of the PR: \
`$ git checkout pull/10247` \
`$ git pull https://git.openjdk.org/jdk pull/10247/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 10247`

View PR using the GUI difftool: \
`$ git pr show -t 10247`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/10247.diff">https://git.openjdk.org/jdk/pull/10247.diff</a>

</details>

### CODE REVIEW

File: src/hotspot/share/utilities/elfFile.cpp [hunk 1]
```diff
@@ -34,6 +34,7 @@
 #include "jvm_io.h"
 #include "logging/log.hpp"
 #include "memory/allocation.inline.hpp"
+#include "memory/resourceArea.hpp"
```
Comment (DamonFool) [inline src/hotspot/share/utilities/elfFile.cpp:37]: Shall we only include `resourceArea.hpp` for `defined(PPC64) && !defined(ABI_ELFv2)`?

Comment (shipilev) [reply src/hotspot/share/utilities/elfFile.cpp:37]: I don't like complicated include guards in include blocks, so I'd prefer not to.

Comment (kimbarrett) [reply src/hotspot/share/utilities/elfFile.cpp:37]: I agree with @shipilev on this.

**Review by kimbarrett — APPROVED** [review]

**Review by TheRealMDoerr — APPROVED** [review]
Thanks for fixing! Build has worked for us without this fix because of PCH.

**Comment by tstuefe** [comment]
I removed the include because we should not use RA in these functions. Please do not re-add it. See https://github.com/openjdk/jdk/pull/10158 . 

Looks like I just forgot to fix up PPCBE. The better fix would be not to use RA.

**Comment by TheRealMDoerr** [comment]
Oh, seems like the `ResourceMark` is no longer needed. At least, I can't see anything left which would need it after https://github.com/openjdk/jdk/pull/10158.

**Comment by shipilev** [comment]
Oh, I see. Let's remove `ResourceMark` then, like in new commit? I don't have a PPC64BE machine to test it, though.

**Comment by shipilev** [comment]
> @kimbarrett Only the author (@shipilev) is allowed to issue the `reviewer` command.

Ah-ha, gotcha! Welcome to PPC64 maintainers team, @kimbarrett! ;)

/reviewer remove @kimbarrett

**Comment by shipilev** [comment]
> @shipilev There are no manually specified reviewers associated with this pull request. Current credited reviewers are:

Nevermind, I think bots have updated the reviewer list, based on "Approved" -> "Comment" change.

**Review by tstuefe — APPROVED** [review]
Good and trivial.

Thanks, Thomas

**Review by TheRealMDoerr — APPROVED** [review]

**Comment by shipilev** [comment]
Thanks!

/integrate

**Pull request closed without merging**